substrate/frame/support/src/traits/messages.rs

```diff
@@ -174,6 +176,61 @@ pub struct BatchFootprint {
 	pub new_pages_count: u32,
 }
 
+/// The resource footprints of continuous subsets of messages.
+///
+/// For a set of messages `xcms[0..n]`, each `footprints[i]` contains the footprint
+/// of the batch `xcms[0..i]`, so as `i` increases `footprints[i]` contains the footprint
+/// of a bigger batch.
+#[derive(Default, RuntimeDebug)]
+pub struct BatchesFootprints {
+	pub first_page_pos: usize,
+	pub footprints: Vec<BatchFootprint>,
```

Comment:
@serban300 please add here some docs, because I am not sure if `first_page_pos` is clear enough, does it mean something like "page_pos where the `search_best_by` batch will be placed"?

Reply:
Done.

> does it mean something like "page_pos where the search_best_by batch will be placed"

Yes, exactly.